lib/Common/ConfigFlagsList.h

```diff
@@ -387,6 +387,7 @@ PHASE(All)
 #else
     #define DEFAULT_CONFIG_WasmFastArray    (false)
 #endif
+#define DEFAULT_CONFIG_WASM_FOLD          (false)
```

Comment:
@MikeHolman 

None, AFAIK. 

* W/ the flag always on, there are quite a few existing unit tests (22+ files) that trigger constant folding which provides me w/ some level of confidence in the functional correctness of this opt.
* Also, I didn't observe any apparent perf regressions on emscripten microbenchmarks I ran. Didn't see any breath-taking improvements either since emscripten benches seem to expose very few opportunities.

I simply added the flag to err on the side of caution.